### PR TITLE
Fix/adapting date formats

### LINKED ISSUE #318: Adapting date formats for internationalization using locales
Currently, dates are displayed in a basic format following the U.S. calendar. We need to change all these sections to be adjusted using locales.

Its possible get current Locale using the hook 

`const locale = useLocale()`

Search what fits better: `fns` `moment` or `luxon`

image

image

### PR DESCRIPTION
**Pull request by syrok94**
**What does this PR do?**
fix #318 

**Related Issue(s)**
none

**Changes**
added a hook useLocale.

**By submitting this PR, I confirm that:**
- [x] I have reviewed my code and believe it is ready for merging.
- [x] I understand that this PR may be subject to review and changes.
- [x] I agree to abide by the code of conduct and contributing guidelines of this project.

### CODE REVIEW

File: src/components/cubes/TableRow.tsx [hunk 1]
```diff
@@ -78,7 +71,7 @@ export default function TableRow({ cube }: { cube: Cube }) {
           {`${cube?.solves?.session.length}/${cube?.solves?.all.length}`}
         </div>
         <div className="hidden text-center align-middle cursor-pointer md:table-cell">
-          {formatDate(cube?.createdAt)}
+          {locale}
```
Comment (bryanlundberg) [inline src/components/cubes/TableRow.tsx:74]: I think this variable name does not describe its purpose.

File: src/hooks/useLocale.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { DateTime } from "luxon";
+import { useEffect, useState } from "react";
+
+export default function useLocale(time: number | undefined): string | null {
+  const [formattedDate, setFormattedDate] = useState<string>("");
+
+  useEffect(() => {
+    if (time) {
+      const parsedDate = DateTime.fromMillis(time)
+        .setLocale(navigator.language)
```
Comment (bryanlundberg) [inline src/hooks/useLocale.ts:10]: Locales should be based on the user's selection in the app settings. In this case, you are using the navigator's language.

You can get the current app locale using the hook `const locale = useLocale()` from `next-intl`.

**Review by bryanlundberg — CHANGES_REQUESTED** [review]
Hi @syrok94 Thank you for your work on this PR!

I've reviewed the changes and left some comments for your consideration. I noticed there is an issue with the overall task execution. Each language has a specific date structure.

For example:

English: MM/DD/YYYY
Spanish: DD/MM/YYYY
Japanese: YYYY/MM/DD

and so on...

The idea is to perform the conversion into the correct format depending on the user's language.

Additionally, I was expecting the library to translate the month names.

image


Sadly I was not able to see these changes.


- Additionally, it would be great if you could implement it as a utility function instead of a hook.

Please address the minor points I've mentioned. 👋👋

**Comment by syrok94** [comment]
Sure, will change accordingly

**Pull request closed without merging**